paasta_tools/cli/cmds/local_run.py

```diff
@@ -632,18 +632,25 @@ def configure_and_run_docker_container(
     Function prints the output of run command in stdout.
     """
 
-    soa_dir = args.yelpsoa_config_root
+    if instance is None and args.healthcheck:
+        paasta_print(
+            "With --healthcheck, --instance must be provided!",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+    if instance is None and not sys.stdin.isatty():
```

Comment:
@solarkennedy it would be nice to have a different error message between no `--instance` (you need to specify `--instance` and `--cluster`) vs interactive being set with `-I` (don't use `-I` without a tty). We can't have that if we just check `interactive is True and not sys.stdin.isatty()`.